components/layout_2020/flow/inline.rs

```diff
@@ -77,6 +81,119 @@ struct LinesBoxes {
+                                    }
+                                };
+                            }
+
+                            add!(first_fragment, inline_start);
+                            self.traverse(layout_context, &inline_box.children);
+                            add!(last_fragment, inline_end);
+                        },
+                        InlineLevelBox::TextRun(text_run) => {
+                            let (_, _, _, runs, break_at_start) =
+                                text_run.break_and_shape(layout_context);
+                            if break_at_start {
+                                self.line_break_opportunity()
+                            }
+                            for run in &runs {
+                                let advance = Length::from(run.glyph_store.total_advance());
+                                if run.glyph_store.is_whitespace() {
+                                    self.line_break_opportunity()
+                                } else {
+                                    self.current_line.min_content += advance
+                                }
+                                self.current_line.max_content += advance
+                            }
+                        },
+                        InlineLevelBox::Atomic(atomic) => {
+                            let inner = || {
+                                // atomic
+                                // .intrinsic_inline_sizes
+                                // .as_ref()
+                                // .expect("Accessing intrinsic size that was not requested")
+                                // .clone()
+                                todo!()
+                            };
+                            let (outer, pc) = outer_intrinsic_inline_sizes(&atomic.style, &inner);
+                            self.current_line.min_content += outer.min_content;
+                            self.current_line.max_content += outer.max_content;
+                            self.current_line_percentages += pc;
+                        },
+                        InlineLevelBox::OutOfFlowFloatBox(_) |
+                        InlineLevelBox::OutOfFlowAbsolutelyPositionedBox(_) => {},
+                    }
+                }
+            }
+
+            fn line_break_opportunity(&mut self) {
+                self.paragraph
+                    .min_content
+                    .max_assign(take(&mut self.current_line.min_content));
+            }
+
+            fn forced_line_break(&mut self) {
+                self.line_break_opportunity();
+                self.current_line
+                    .adjust_for_pbm_percentages(take(&mut self.current_line_percentages));
+                self.paragraph
+                    .max_content
+                    .max_assign(take(&mut self.current_line.max_content));
+            }
+        }
+        fn take<T: Zero>(x: &mut T) -> T {
+            std::mem::replace(x, T::zero())
+        }
```

Comment:
Don't we only call that twice?

Reply:
Three times. I found it made the code around the calls easier to follow (especially since it’s a bit repetitive).